app/module/App/src/Controller/StatisticsController.php

```diff
@@ -62,19 +62,27 @@ public function __construct(
     public function indexAction(array $params)
     {
         try {
+            $time_format = 'g:i:s';
+            error_log('Starting statistics calculation ' . date($time_format));
+
             $startDate = $this->extractDate($params['start_date'] ?? null);
             $endDate   = $this->extractDate($params['end_date'] ?? null);
             $params    = ParamsBuilder::reportStatsParams($startDate, $endDate);
 
+
             $posts = $this->socialService->fetchPosts();
+            error_log('Fetched stats ' . date($time_format));
```

Comment:
Decided to add some logs since I have noticed that the code is quite slow.

It's clear that the slow part is in this application. I tried to narrow it down. Even with just one calculation the speed is about the same. I think the data sample is pretty solid in size. I think it's due to memory allocation and consumption. PHP is famous to being memory thirsty. I would be interested to know why this app is so slow.